packages/workflows-shared/src/binding.ts

```diff
@@ -34,22 +31,47 @@ export class WorkflowBinding extends WorkerEntrypoint<Env> implements Workflow {
 			}
 		);
 
-		return new WorkflowHandle(id, stub);
+		const handle = new WorkflowHandle(id, stub);
+		return {
+			id: id,
+			pause: handle.pause.bind(handle),
+			resume: handle.resume.bind(handle),
+			terminate: handle.terminate.bind(handle),
+			restart: handle.restart.bind(handle),
+			status: handle.status.bind(handle),
+		};
 	}
 
 	public async get(id: string): Promise<WorkflowInstance> {
-		const stubId = this.env.ENGINE.idFromName(id);
-		const stub = this.env.ENGINE.get(stubId);
-		return new WorkflowHandle(id, stub);
+		const engineStubId = this.env.ENGINE.idFromName(id);
+		const engineStub = this.env.ENGINE.get(engineStubId);
+
+		const handle = new WorkflowHandle(id, engineStub);
+
+		try {
+			await handle.status();
+		} catch (e) {
+			throw new Error("instance.not_found");
+		}
+
+		return {
+			id: id,
+			pause: handle.pause.bind(handle),
+			resume: handle.resume.bind(handle),
+			terminate: handle.terminate.bind(handle),
+			restart: handle.restart.bind(handle),
+			status: handle.status.bind(handle),
+		};
 	}
 }
 
 export class WorkflowHandle extends RpcTarget implements WorkflowInstance {
-	constructor(
-		public id: string,
-		private stub: DurableObjectStub<Engine>
-	) {
+	public id: string;
+	private stub: DurableObjectStub<Engine>;
+	constructor(id: string, stub: DurableObjectStub<Engine>) {
 		super();
+		this.id = id;
+		this.stub = stub;
```

Comment:
Why this change? This seems like equivalent code to me

Reply:
I'd lean towards re-introducing it if possible for readability—if it's causing bugs we should investigate